ruby/lib/ci/queue/common.rb

```diff
@@ -42,14 +42,24 @@ def ordering_strategy
         when :timing_based
           Strategy::TimingBased.new(config)
         when :suite_bin_packing
-          # pass redis if available
-          # need to think about a better way to structure queue/strategy interaction
-          redis_instance = if self.respond_to?(:redis, true) # include private methods
-            self.send(:redis)
+          # Use dedicated timing redis if provided via CLI flag (stored in config).
+          # Do NOT fall back to the default queue redis for timing reads.
+          timing_url = if config.respond_to?(:timing_redis_url)
+            config.timing_redis_url
           else
             nil
           end
-          Strategy::SuiteBinPacking.new(config, redis: redis_instance)
+          timing_redis = nil
+          if timing_url && !timing_url.empty?
+            begin
+              require 'redis'
```

Comment:
this is kind of an antipattern, can we just always require redis at the top? presumably we have it already in memory or will for other redis usage